instrumentation/armeria-1.3/library/src/main/java/io/opentelemetry/instrumentation/armeria/v1_3/ArmeriaHttpAttributesExtractor.java

```diff
@@ -53,28 +53,36 @@ protected String userAgent(RequestContext ctx) {
   }
 
   @Override
-  protected Long requestContentLength(RequestContext ctx, RequestLog requestLog) {
+  @Nullable
+  protected Long requestContentLength(RequestContext ctx, @Nullable RequestLog requestLog) {
+    if (requestLog == null) {
+      return null;
+    }
     return requestLog.requestLength();
   }
 
   @Override
   @Nullable
-  protected Long requestContentLengthUncompressed(RequestContext ctx, RequestLog requestLog) {
+  protected Long requestContentLengthUncompressed(
+      RequestContext ctx, @Nullable RequestLog requestLog) {
     return null;
   }
 
   @Override
   @Nullable
-  protected Long statusCode(RequestContext ctx, RequestLog requestLog) {
+  protected Long statusCode(RequestContext ctx, @Nullable RequestLog requestLog) {
+    if (requestLog == null) {
+      return null;
+    }
     HttpStatus status = requestLog.responseHeaders().status();
-    if (!status.equals(HttpStatus.UNKNOWN)) {
-      return (long) status.code();
+    if (status.equals(HttpStatus.UNKNOWN)) {
+      return null;
     }
-    return null;
+    return (long) status.code();
   }
 
   @Override
-  protected String flavor(RequestContext ctx, RequestLog requestLog) {
+  protected String flavor(RequestContext ctx, @Nullable RequestLog requestLog) {
```

Comment:
Hmm - I'd like to not leak `@Nullable` in when it's not null. I guess what we want is to "only allow null if Void". But not sure we can model that, so maybe it's ok as it is, does it cause any nullability issues right now?

Reply:
I think most instrumentations (other than armeria) will also pass `null` response to `end()` when exception occurs

Reply:
Ah thanks - in that case yeah skipping the invocation if null seems easier for instrumentation

Reply:
added skipping of calls to

* statusCode()
* responseContentLength()
* responseContentLengthUncompressed()

when response is null